skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceCallback.scala

```diff
@@ -0,0 +1,267 @@
+    committed1(partitions, none)
+
+  def committed[F[_]: MonadThrowable](
+    partitions: Nes[TopicPartition],
+    timeout: FiniteDuration
+  ): RebalanceCallback[F, Map[TopicPartition, OffsetAndMetadata]] =
+    committed1(partitions, timeout.some)
+
+  def endOffsets[F[_]: MonadThrowable](
+    partitions: Nes[TopicPartition]
+  ): RebalanceCallback[F, Map[TopicPartition, Offset]] =
+    offsets1(_.endOffsets(partitions.asJava))
+
+  def endOffsets[F[_]: MonadThrowable](
+    partitions: Nes[TopicPartition],
+    timeout: FiniteDuration
+  ): RebalanceCallback[F, Map[TopicPartition, Offset]] =
+    offsets1(_.endOffsets(partitions.asJava, timeout.asJava))
+
+  def groupMetadata[F[_]]: RebalanceCallback[F, ConsumerGroupMetadata] =
+    ConsumerStep(_.groupMetadata().asScala)
+
+  def topics[F[_]: MonadThrowable]: RebalanceCallback[F, Map[Topic, List[PartitionInfo]]] =
+    for {
+      result <- ConsumerStep(_.listTopics())
+      result <- lift(partitionsInfoMapF[F](result))
+    } yield result
+
+  def topics[F[_]: MonadThrowable](timeout: FiniteDuration): RebalanceCallback[F, Map[Topic, List[PartitionInfo]]] =
+    for {
+      result <- ConsumerStep(_.listTopics(timeout.asJava))
+      result <- lift(partitionsInfoMapF[F](result))
+    } yield result
+
+  def offsetsForTimes[F[_]: MonadThrowable](
+    timestampsToSearch: Nem[TopicPartition, Instant]
+  ): RebalanceCallback[F, Map[TopicPartition, Option[OffsetAndTimestamp]]] =
+    for {
+      result <- ConsumerStep(_.offsetsForTimes(timestampsToSearchJ(timestampsToSearch)))
+      result <- lift(offsetsAndTimestampsMapF[F](result))
+    } yield result
+
+  def offsetsForTimes[F[_]: MonadThrowable](
+    timestampsToSearch: Nem[TopicPartition, Instant],
+    timeout: FiniteDuration
+  ): RebalanceCallback[F, Map[TopicPartition, Option[OffsetAndTimestamp]]] =
+    for {
+      result <- ConsumerStep(_.offsetsForTimes(timestampsToSearchJ(timestampsToSearch), timeout.asJava))
+      result <- lift(offsetsAndTimestampsMapF[F](result))
+    } yield result
+
+  def partitionsFor[F[_]: ApplicativeThrowable](topic: Topic): RebalanceCallback[F, List[PartitionInfo]] =
+    for {
+      result <- ConsumerStep(_.partitionsFor(topic))
+      result <- lift(partitionsInfoListF[F](result))
+    } yield result
+
+  def partitionsFor[F[_]: ApplicativeThrowable](
+    topic: Topic,
+    timeout: FiniteDuration
+  ): RebalanceCallback[F, List[PartitionInfo]] =
+    for {
+      result <- ConsumerStep(_.partitionsFor(topic, timeout.asJava))
+      result <- lift(partitionsInfoListF[F](result))
+    } yield result
+
+  def paused[F[_]]: RebalanceCallback[F, Set[TopicPartition]] =
+    ConsumerStep(
+      _.paused().asScala.toSet.map(unsafeTopicPartition)
+    )
+
+  // TODO using Offset.unsafe to avoid requiring ApplicativeThrowable as in Offset.of
+  //  also it looks like consumer.position cannot return invalid offset
+  def position[F[_]](tp: TopicPartition): RebalanceCallback[F, Offset] =
```

Comment:
none of consumer related methods should require `F`, period.

Reply:
done in 3654a057